pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -410,3 +373,123 @@ func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStor
 		}
 	}
 }
+
+func fetchAndStoreGuestInfo(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	domain, err := con.LookupDomainByName(domainName)
+	if err != nil {
+		// Skip execution on error, domain lookup failed
+		return
+	}
+	defer domain.Free()
+
+	flags := libvirt.DOMAIN_GUEST_INFO_INTERFACES |
+		libvirt.DOMAIN_GUEST_INFO_OS |
+		libvirt.DOMAIN_GUEST_INFO_HOSTNAME |
+		libvirt.DOMAIN_GUEST_INFO_TIMEZONE |
+		libvirt.DOMAIN_GUEST_INFO_USERS
+
+	guestInfo, err := domain.GetGuestInfo(flags, 0)
+	if err != nil {
+		// If fetching guest info fails, skip processing
+		return
+	}
+
+	for _, command := range commands {
```

Comment:
Depending on the answer to the above comment, if there is no reason to have different polling intervals, you can remove this for loop and store all values.